.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,82 @@
 Hi! Before submitting your PR for review, and/or before merging it, please
-go through the following checklist:
+go through the following checklists:
 
-- [ ] For DB changes, check for missing indexes, check to see if the changes
-affect other apps (such as the dashboard), make sure the DB columns in the
-various environments are properly populated, coordinate with devops, plan
-migrations in separate steps.
+### General
 
-- [ ] For route changes, make sure GET requests don't change state or result in
-destructive behavior. GET requests should only result in information being
-read, not written.
+- [ ] `make lint` ran locally without errors
 
-- [ ] For encryption changes, make sure it is compatible with data that was
-encrypted with the old code.
+- [ ] Rubocop or Reek offenses were not disabled unless they are false positives.
+If you're not sure, please ask a teammate.
 
-- [ ] For secrets changes, [make sure to update the S3 secrets bucket](https://github.com/18F/identity-private/wiki/Secrets-S3-buckets) with the 
-new configs in **all** environments. 
+- [ ] Tests added for this feature/bug
 
-- [ ] Do not disable Rubocop or Reek offenses unless you are absolutely sure
-they are false positives. If you're not sure how to fix the offense, please
-ask a teammate.
+- [ ] The PR adheres to our [PR guidelines]
+
+- [ ] Related commits have been squashed
+
+- [ ] Method names in classes are in descending order of abstraction.
```

Comment:
I wouldn't see this as a hard and fast rule, but a general rule that should apply to related methods. So, for example, I could see as reasonable grouping a set of methods that together accomplish a task, and ordering them by decreasing level of abstraction. As opposed to mixing everything up so we get a strict ordering across all methods in the class, but now have to constantly hop around everywhere to get in our minds the few methods we need to understand the flow.

Reply:
Yes, that is the intention. I should clarify that this mainly applies to classes like service objects that have one public method that might then call several private methods, as shown in the video I linked to.